### PR TITLE
App/Data: Implement a DAO for the ML Model entity 

### PR DESCRIPTION
**Pull request by wooksong**
This patch adds the initial implementation of a DAO for the ML Model entity.

Signed-off-by: Wook Song <wook16.song@samsung.com>

### CODE REVIEW

File: ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/data/OfflineModelsRepository.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package ai.nnstreamer.ml.inference.offloading.data
+
+import kotlinx.coroutines.flow.Flow
+
+
+class OfflineModelsRepository(private val modelDao: ModelDao) : ModelsRepository {
```
Comment (niley7464) [inline ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/data/OfflineModelsRepository.kt:6]: It's just a question, do you have a plan to make other ModelsRepository class? (OnlineModelsRepository..?)

**Review by niley7464 — APPROVED** [review]
LGTM!

File: ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/data/OfflineModelsRepository.kt (hunk 1 above)
Comment (wooksong) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/data/OfflineModelsRepository.kt:6]: Ah, the term 'Repository' is used here in different contexts. Please refer to this [article](https://developer.android.com/topic/architecture/data-layer/offline-first#reads).

Comment (niley7464) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/data/OfflineModelsRepository.kt:6]: Oh, I thought using interface is for another class like online.
There is a lot of content in the article that I am seeing for the first time.
Thank you :)

**Pull request merged**